.github/workflows/analyze-pr-changes.yml

```diff
@@ -0,0 +1,168 @@
+name: Analyze PR Changes
+
+on: [pull_request]
+
+env:
+  OPENSEARCH_INITIAL_ADMIN_PASSWORD: BobgG7YrtsdKf9M
+
+jobs:
+  determine-changes:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Build Docker Image
+        run: docker build coverage --tag opensearch-with-api-plugin
+
+      - name: Run OpenSearch
+        shell: bash -eo pipefail {0}
+        run: |
+          docker run \
+            --name opensearch \
+            --rm -d \
+            -p 9200:9200 -p 9600:9600 \
+            -e "discovery.type=single-node" \
+            -e OPENSEARCH_INITIAL_ADMIN_PASSWORD="$OPENSEARCH_INITIAL_ADMIN_PASSWORD" \
+            opensearch-with-api-plugin
+          
+          for attempt in {1..20}; do
+            sleep 5
+            if curl -ksS -u "admin:$OPENSEARCH_INITIAL_ADMIN_PASSWORD" https://localhost:9200; then
+              echo '=> ready'
+              exit 0
+            fi
+            echo '=> waiting...'
+          done
+          exit 1
```

Comment:
Since we're writing tools here, would add a JS tool to wait for OpenSearch to start that polls.